src/Native/Arena.ts

```diff
@@ -104,22 +101,23 @@ export default class ArenaEntity extends Entity implements TeamGroupEntity {
 
         this.arenaData.values.flags = ArenaFlags.gameReadyStart;
         this.arenaData.values.playersNeeded = 0;
-        this.arenaData.values.ticksUntilStart = this.COUNTDOWN_TICKS;
+        this.arenaData.values.ticksUntilStart = countdownTicks;
 
         this.teamData.values.teamColor = Color.Neutral;
     }
 
     /**
      * Finds a spawnable location on the map.
      */
-     public findSpawnLocation(): VectorAbstract {
+     public findSpawnLocation(isPlayer: boolean=false): VectorAbstract {
         const pos = {
             x: ~~(Math.random() * this.width - this.width / 2),
             y: ~~(Math.random() * this.height - this.height / 2),
         }
 
         findSpawn: for (let i = 0; i < 20; ++i) {
-            if (!this.isValidSpawnLocation(pos.x, pos.y)) {
+            if (!this.isValidSpawnLocation(pos.x, pos.y) ||
+                isPlayer && Math.max(pos.x, pos.y) < this.arenaData.values.rightX / 5 && Math.min(pos.x, pos.y) > this.arenaData.values.leftX / 5) {
```

Comment:
1. this isn't how it works in diep.io - why is this here?
2. doesn't work for rectangular maps